src/test/regress/expected/failure_create_distributed_table_non_empty.out

```diff
@@ -0,0 +1,520 @@
+CREATE TABLE test_table(id int, value_1 int);
+INSERT INTO test_table VALUES (1,1),(2,2),(3,3),(4,4);
+-- in the first test, kill the first connection we sent from the coordinator
+SELECT citus.mitmproxy('conn.kill()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+SELECT create_distributed_table('test_table', 'id');
+ERROR:  connection error: localhost:57640
+DETAIL:  server closed the connection unexpectedly
+	This probably means the server terminated abnormally
+	before or while processing the request.
+SELECT citus.mitmproxy('conn.allow()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+SELECT count(*) FROM pg_dist_shard WHERE logicalrelid='create_distributed_table_non_empty_failure.test_table'::regclass;
+ count 
+-------
+     0
+(1 row)
+
+-- in the first test, cancel the first connection we sent from the coordinator
+SELECT citus.mitmproxy('conn.cancel(' ||  pg_backend_pid() || ')');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+SELECT create_distributed_table('test_table', 'id');
+ERROR:  canceling statement due to user request
+SELECT citus.mitmproxy('conn.allow()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+SELECT count(*) FROM pg_dist_shard WHERE logicalrelid='create_distributed_table_non_empty_failure.test_table'::regclass;
+ count 
+-------
+     0
+(1 row)
+
+-- kill as soon as the coordinator sends CREATE SCHEMA
+SELECT citus.mitmproxy('conn.onQuery(query="^CREATE SCHEMA").kill()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+SELECT create_distributed_table('test_table', 'id');
+ERROR:  server closed the connection unexpectedly
+	This probably means the server terminated abnormally
+	before or while processing the request.
+CONTEXT:  while executing command on localhost:57640
+SELECT citus.mitmproxy('conn.allow()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+SELECT count(*) FROM pg_dist_shard WHERE logicalrelid='create_distributed_table_non_empty_failure.test_table'::regclass;
+ count 
+-------
+     0
+(1 row)
+
+SELECT run_command_on_workers($$SELECT count(*) FROM information_schema.schemata WHERE schema_name = 'create_distributed_table_non_empty_failure'$$);
+ run_command_on_workers 
+------------------------
+ (localhost,57637,t,1)
+ (localhost,57640,t,0)
+(2 rows)
+
+-- cancel as soon as the coordinator sends CREATE SCHEMA
+SELECT citus.mitmproxy('conn.onQuery(query="^CREATE SCHEMA").cancel(' ||  pg_backend_pid() || ')');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+SELECT create_distributed_table('test_table', 'id');
+ERROR:  canceling statement due to user request
+SELECT citus.mitmproxy('conn.allow()');
+ mitmproxy 
+-----------
+ 
+(1 row)
+
+SELECT count(*) FROM pg_dist_shard WHERE logicalrelid='create_distributed_table_non_empty_failure.test_table'::regclass;
+ count 
+-------
+     0
+(1 row)
+
+SELECT run_command_on_workers($$SELECT count(*) FROM information_schema.schemata WHERE schema_name = 'create_distributed_table_non_empty_failure'$$);
+ run_command_on_workers 
+------------------------
+ (localhost,57637,t,1)
+ (localhost,57640,t,1)
```

Comment:
But now the schema exists on both, even though the user cancelled. Is that a bug?

Reply:
As I understand, for `cancel`, there is a small amount of time that the coordinator gets aware of the `cancel` operation. During that time, coordinator sends the other `CREATE SCHEMA IF NOT EXISTS` command. I think the only reason we hit to that issue is that we make the `EnsureSchemaExists` call before starting the coordinated transaction.

Reply:
Hmm, this might be counted as a bug. It turns out that Citus calls `CHECK_FOR_INTERRUPTS` on `GetRemoteCommandResult()`. This means that the command has already been sent and (via `SendRemoteCommand`) and the schema has already been created when the cancel is processed.

**I think this is not a problem for queries that are inside the coordinated transaction because we'd already be rollbacking those. @furkansahin and @lithp do you agree with me on this?**

See the bt when I sent the `SIGINT` just before calling `ExecuteCriticalRemoteCommand()`

```SQL
Program received signal SIGINT, Interrupt.

Breakpoint 2, errstart (elevel=20, filename=0x10a48a678 "postgres.c", lineno=2997, funcname=0x10a48a938 "ProcessInterrupts", domain=0x0) at elog.c:251
251			if (CritSectionCount > 0)
(gdb) bt
#0  errstart (elevel=20, filename=0x10a48a678 "postgres.c", lineno=2997, funcname=0x10a48a938 "ProcessInterrupts", domain=0x0) at elog.c:251
#1  0x000000010a0d22be in ProcessInterrupts () at postgres.c:2995
#2  0x000000010a9ccd9d in FinishConnectionIO (connection=0x7f913183cd28, raiseInterrupts=1 '\001') at connection/remote_commands.c:673
#3  0x000000010a9cc091 in GetRemoteCommandResult (connection=0x7f913183cd28, raiseInterrupts=1 '\001') at connection/remote_commands.c:550
#4  0x000000010a9cc948 in ExecuteCriticalRemoteCommand (connection=0x7f913183cd28, command=0x7f913185f930 "CREATE SCHEMA IF NOT EXISTS x AUTHORIZATION onderkalaci") at connection/remote_commands.c:372
#5  0x000000010a9c1447 in EnsureSchemaExistsOnAllNodes (relationId=16824) at commands/create_distributed_table.c:862
#6  0x000000010a9c22bf in CreateHashDistributedTableShards (relationId=16824, colocatedTableId=16820, localTableEmpty=1 '\001') at commands/create_distributed_table.c:489
#7  0x000000010a9c0f53 in CreateDistributedTable (relationId=16824, distributionColumn=0x7f913101da40, distributionMethod=104 'h', colocateWithTableName=0x7f913101e498 "default", viaDeprecatedAPI=0 '\0') 
....
```

But, we should probably discuss this with @marcocitus to understand why `SendRemoteCommand` doesn't check for interrupts? I'm OK with opening an issue for this.

But, in any case, @furkansahin could you add a comment to the test?

Reply:
> I think this is not a problem for queries that are inside the coordinated transaction because we'd already be rollbacking those. @furkansahin and @lithp do you agree with me on this?

Yes, I agree.